rust/src/pet/message/round.rs

```diff
@@ -0,0 +1,142 @@
+#![allow(dead_code)] // temporary
+
+use std::ops::Range;
+
+use sodiumoxide::crypto::{box_, sealedbox, sign};
+
+use super::{ROUND_TAG, TAG_RANGE};
+use crate::pet::PetError;
+
+// round box field ranges
+const ENCR_PK_RANGE: Range<usize> = 1..33; // 32 bytes
+const SIGN_PK_RANGE: Range<usize> = 33..65; // 32 bytes
+
+/// Mutable and immutable buffer access to round box fields.
+struct RoundBoxBuffer<B> {
+    bytes: B,
+}
+
+impl RoundBoxBuffer<Vec<u8>> {
+    /// Create an empty round box buffer of size `len`.
+    fn new(len: usize) -> Self {
+        Self {
+            bytes: vec![0_u8; len],
+        }
+    }
+}
+
+impl<B: AsRef<[u8]>> RoundBoxBuffer<B> {
+    /// Create a round box buffer from `bytes`. Fails if the `bytes` don't conform to the expected
+    /// round box length `exp_len`.
+    fn from(bytes: B, exp_len: usize) -> Result<Self, PetError> {
```

Comment:
Why do we have to pass `exp_len` here? Don't we already know that it's 65 bytes? It seems a bit weird to have the caller pass the expected length, because if they already know it, they do the check themself.

Reply:
my intention was to have a similar buffer interface for for `round` as for `sum`/`update`/`sum2`. the former could go without the `exp_len`, but the latter need this (update already now, sum/sum2 as soon as certificates are not dummy anymore).

since the `Buffer`s are only privately available to the `Box`es, there is no risk in wrongly calling them from the outside. not sure what's the best approach here.